azure/scope/managedcontrolplane_test.go

```diff
@@ -1619,3 +1621,156 @@ func TestManagedControlPlaneScope_AutoUpgradeProfile(t *testing.T) {
 		})
 	}
 }
+
+func TestManagedControlPlaneScope_GroupSpecs(t *testing.T) {
+	cases := []struct {
+		name     string
+		input    ManagedControlPlaneScopeParams
+		expected []azure.ASOResourceSpecGetter[*asoresourcesv1.ResourceGroup]
+	}{
+		{
+			name: "virtualNetwork belongs to a different resource group",
+			input: ManagedControlPlaneScopeParams{
+				Cluster: &clusterv1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "cluster1",
+						Namespace: "default",
+					},
+				},
+				ControlPlane: &infrav1.AzureManagedControlPlane{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "cluster1",
+						Namespace: "default",
+					},
+					Spec: infrav1.AzureManagedControlPlaneSpec{
+						ResourceGroupName: "dummy-rg",
+						AzureManagedControlPlaneClassSpec: infrav1.AzureManagedControlPlaneClassSpec{
+							Location: "my-location",
+							VirtualNetwork: infrav1.ManagedControlPlaneVirtualNetwork{
+								ResourceGroup: "different-rg",
+								ManagedControlPlaneVirtualNetworkClassSpec: infrav1.ManagedControlPlaneVirtualNetworkClassSpec{
+									Name:      "vnet1",
+									CIDRBlock: "10.1.0.0/8",
+									Subnet: infrav1.ManagedControlPlaneSubnet{
+										Name:      "subnet1",
+										CIDRBlock: "10.1.0.0/16",
+									},
+								},
+							},
```

Comment:
If only `ResourceGroup` has any bearing on the result of the test, I'd vote to remove everything else in each test case.